src/plugins/organisation/PMRWindow/src/pmrwindowwindow.cpp

```diff
@@ -99,19 +85,29 @@ PmrWindowWindow::PmrWindowWindow(QWidget *pParent) :
     connect(this, SIGNAL(visibilityChanged(bool)),
             this, SLOT(retrieveExposuresList(const bool &)));
 
-    // Create a network access manager so that we can then retrieve various
-    // things from the PMR
+    // Create and add the PMR web service
+
+    mPmrWebService = new PMRSupport::PmrWebService();
+
+    // Some connections to process responses from the web service
 
-    mNetworkAccessManager = new QNetworkAccessManager(this);
+    connect(mPmrWebService, SIGNAL(busy(const bool &)),
+            this, SLOT(busy(const bool &)));
 
-    // Make sure that we get told when our PMR requests are finished, as well as
-    // if there are SSL errors (which would happen if the website's certificate
-    // is invalid, e.g. it has expired)
+    connect(mPmrWebService, SIGNAL(showWarning(const QString &)),
+            this, SLOT(showInformation(const QString &)));
+    connect(mPmrWebService, SIGNAL(showWarning(const QString &, const QString &)),
+            this, SLOT(showInformation(const QString &, const QString &)));
```

Comment:
Why two signals ```showWarning()```? Couldn't you have only one of them with the second parameter defaulted to ```QString()```?

Also, why call the signal ```showWarning()``` when your slot is called ```showInformation()```?